### DISCUSSION TITLE
GraphQL: Page - singleByPath

### DISCUSSION THREAD

**Discussion by MRVDOG** [requarks/wiki#6672] [Error / Bug Report] [answered]
Hi, I want to fetch a page by its path so I can load some of the data into our app's dashboard, This page is public, so I only allowed read-only access with my API token, The issue I am running into is, according to this code 
https://github.com/requarks/wiki/blob/cd390ec54472f2678243ee8ccc3822333a09fa1c/server/graph/resolvers/page.js#L152-L196
the `singleByPath` query requires `manage:pages` and `delete:pages` permissions, I'm a little confused as to why these perms are required when it just returns the page data, managing/editing/deleting would be a Mutation, not a Query, so the methods in PageQuery, should in theory only require read access 🤔

**Comment by NGPixel** [comment, marked as answer]
This is been addressed in v3.

The GraphQL API in v2 was not originally meant for external use.

**Reply by MRVDOG** [reply]
awesome thanks, any ETA on v3? or still just h2 2023?
